packages/core-components/api-report.md

```diff
@@ -57,6 +59,45 @@ enum Alignment {
   UP_RIGHT = 'UR',
 }
 
+// Warning: (ae-missing-release-tag) "AsyncFallbackOptions" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export interface AsyncFallbackOptions {
+  // (undocumented)
+  error?:
+    | JSX.Element
+    | ComponentType<{
+        error: Error;
+      }>;
+  // (undocumented)
+  progress?: JSX.Element | ComponentType<{}>;
+}
+
+// Warning: (ae-missing-release-tag) "AsyncFallbackResult" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export type AsyncFallbackResult<T> =
+  | {
+      fallback: undefined;
+      value: T;
+    }
+  | {
+      fallback: JSX.Element;
+      value: undefined;
+    };
+
+// Warning: (ae-missing-release-tag) "AsyncState" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export interface AsyncState<T> {
+  // (undocumented)
+  error?: Error | undefined;
+  // (undocumented)
+  loading?: boolean | undefined;
+  // (undocumented)
+  value?: T | undefined;
+}
```

Comment:
the actual AsyncState declaration in react-use is richer and arguably more useful:

```ts
export declare type AsyncState<T> = {
    loading: boolean;
    error?: undefined;
    value?: undefined;
} | {
    loading: true;
    error?: Error | undefined;
    value?: T;
} | {
    loading: false;
    error: Error;
    value?: undefined;
} | {
    loading: false;
    error?: undefined;
    value: T;
};
```

Reply:
Except the first in the union which almost always matches and allows value to be undefined.

Anyway, this is _good_. This `AsyncState` must be less-than-or-equally strict, so that you can pass such a state into `useAsyncState`. It also allows damaged types (where e.g. one of these properties are removed) or types that aren't even from `react-use`.

What's stricter is the _return value_ from these hooks, which is of [this type](https://github.com/backstage/backstage/pull/8084/files/7c2f7e548dffd97a8443fc3012fc018964ecf410#diff-ea569eb8425f004ed49a86194d7b5b19b5dff946ced5c6f77933d48edbb0f036R79-R87):

```ts
export type AsyncFallbackResult<T> =
  | {
      fallback: undefined;
      value: T;
    }
  | {
      fallback: JSX.Element;
      value: undefined;
    };
```